miden-tx/src/tests/kernel_tests/test_note.rs

```diff
@@ -429,3 +429,57 @@ fn test_get_note_serial_number() {
     let serial_number = tx_context.input_notes().get_note(0).note().serial_num();
     assert_eq!(process.stack.get_word(0), serial_number);
 }
+
+#[test]
+fn test_get_inputs_hash() {
+    let tx_context = TransactionContextBuilder::with_standard_account(
+        ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_OFF_CHAIN,
+        ONE,
+    )
+    .with_mock_notes(AssetPreservationStatus::Preserved)
+    .build();
+
+    let code = "
+        use.miden::note
+        
+        begin
+            push.1.2.3.4.1000 mem_storew dropw
+            push.5.6.7.8.1001 mem_storew dropw
+            push.9.10.11.12.1002 mem_storew dropw
+            push.13.14.15.16.1003 mem_storew dropw
+
+            push.5.1000
+            exec.note::compute_inputs_hash
+
+            push.8.1000
+            exec.note::compute_inputs_hash
+
+            push.15.1000
+            exec.note::compute_inputs_hash
+
+            push.0.1000
+            exec.note::compute_inputs_hash
+        end
+    ";
+
+    let process = tx_context.execute_code(code).unwrap();
+    let expected_stack = [
+        Felt::new(0),
+        Felt::new(0),
+        Felt::new(0),
+        Felt::new(0),
+        Felt::new(10300020282439016154),
+        Felt::new(3516596904277416676),
+        Felt::new(11018788508269249672),
+        Felt::new(7921509648524809116),
+        Felt::new(13608701685256682132),
+        Felt::new(16013969809933496273),
+        Felt::new(15720844923951376941),
+        Felt::new(15975159621759139720),
+        Felt::new(12095223039215569196),
+        Felt::new(16902760742589336416),
+        Felt::new(12194156716918087419),
+        Felt::new(2777745863384272413),
+    ];
+    assert_eq!(process.get_stack_state()[0..16], expected_stack);
```

Comment:
Is it possible to compute these dynamically? Or at least write comments about how these were computed?